include/drivers/adc.h

```diff
@@ -134,6 +153,41 @@ struct adc_channel_cfg {
 #endif /* CONFIG_ADC_CONFIGURABLE_INPUTS */
 };
 
+/**
+ * @brief Convert a raw ADC value to millivolts.
+ *
+ * This function performs the necessary conversion to transform a raw
+ * ADC measurement to a voltage in millivolts.
+ *
+ * @param ref_mv the reference voltage used for the measurement, in
+ * millivolts.  This may be from adc_ref_internal() or a known
+ * external reference.
+ *
+ * @param gain the ADC gain configuration used to sample the input
+ *
+ * @param resolution the resolution used to sample the input
+ *
+ * @param valp pointer to the raw measurement value on input, and the
+ * corresponding millivolt value on successful conversion.  If
+ * conversion fails the stored value is left unchanged.
+ *
+ * @retval 0 on successful conversion
+ * @retval -ENOTSUP if the gain is not reversible
+ */
+static inline int adc_convert_millivolt(s32_t ref_mv,
```

Comment:
I would personally name this `adc_convert_mv()`

Reply:
I'd go with something like `adc_raw_to_millivolt()`, to make it a bit more explicit, though I don't know the naming conventions here.